File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/broacast_receiver/RestartAlarmAtBootReceiver.java
package dam.isi.frsf.utn.edu.ar.delivery.broacast_receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import dam.isi.frsf.utn.edu.ar.delivery.activity.MainActivity;

import static dam.isi.frsf.utn.edu.ar.delivery.task.AlarmSetter.configureAlarm;

public class RestartAlarmAtBootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if ("android.intent.action.BOOT_COMPLETED".equals(intent.getAction())) {
            configureAlarm(context);
        }
    }
}

File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/broacast_receiver/RestartAlarmAtBootReceiver.kt
package dam.isi.frsf.utn.edu.ar.delivery.broacast_receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import dam.isi.frsf.utn.edu.ar.delivery.task.AlarmSetter.configureAlarm

class RestartAlarmAtBootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if ("android.intent.action.BOOT_COMPLETED" == intent.action) {
            configureAlarm(context)
        }
    }
}